src/include/sof/spi.h

```diff
@@ -0,0 +1,188 @@
+};
+
+enum {
+	SPI_TRIGGER_START = 0,
+	SPI_TRIGGER_STOP,
+};
+
+enum sof_spi_type {
+	SOF_SPI_INTEL_SLAVE = 0,
+	SOF_SPI_INTEL_MASTER,
+};
+
+/* CTRLR0 */
+/* 00-standard spi; 01-dual spi; 10-quad spi */
+#define SPI_FRAME_FORMAT(x)	((x) << 21)
+#define SPI_DATA_FRAME_SIZE(x)	((x) << 16)
+/* 0-slave txt enabled; 1-slave txt disabled */
+#define SPI_SLV_OE(x)		((x) << 10)
+/* 00-both; 01-transmit only;
+ * 10-receive only; 11-eeprom read
+ */
+#define SPI_TRANSFER_MODE(x)	((x) << 8)
+/* 0-inactive low; 1-inactive high */
+#define SPI_SCPOL(x)		((x) << 7)
+/* 0-first edge capture; 1-one cycle after cs line */
+#define SPI_SCPH(x)		((x) << 6)
+/* 00-moto spi; 01-ti ssp; 10-ns microwire */
+#define SPI_FRAME_TYPE(x)	((x) << 4)
+
+/* SSIENR */
+#define SPI_SSIEN		1
+
+/* IMR */
+/* 0-masked; 1-unmasked;
+ * receive FIFO full interrupt mask/unmask
+ */
+#define SPI_IMR_RXFIM(x)	((x) << 4)
+/* 0-masked; 1-unmasked;
+ * receive FIFO overflow interrupt mask/unmask
+ */
+#define SPI_IMR_RXOIM(x)	((x) << 3)
+/* 0-masked; 1-unmasked;
+ * transmit FIFO overflow interrupt mask/unmask
+ */
+#define SPI_IMR_TXOIM(x)	((x) << 1)
+/* 0-masked; 1-unmasked;
+ * transmit FIFO empty interrupt mask/unmask
+ */
+#define SPI_IMR_TXEIM(x)	((x) << 0)
+
+/* DMACR */
+/* 0-transmit DMA disable; 1-transmit DMA enable */
+#define	SPI_DMACR_TDMAE(x)	((x) << 1)
+/* 0-receive DMA disable; 1-receive DMA enable */
+#define SPI_DMACR_RDMAE(x)	((x) << 0)
+/* DMATDLR/DMARDLR */
+/* transmit data level: 0~255 */
+#define SPI_DMATDLR(x)		((x) << 0)
+/* receive data level: 0~255 */
+#define SPI_DMARDLR(x)		((x) << 0)
+
+/* spi register */
+#define SPI_SLAVE_BASE	0x00080000
+#define SUE_SPI_BASEADDRESS(x)	((x) + SPI_SLAVE_BASE)
+#define	SUE_SPI_REG_CTRLR0	SUE_SPI_BASEADDRESS(0x00)
+#define	SUE_SPI_REG_CTRLR1	SUE_SPI_BASEADDRESS(0x04)
+#define	SUE_SPI_REG_SSIENR	SUE_SPI_BASEADDRESS(0x08)
+#define	SUE_SPI_REG_MWCR	SUE_SPI_BASEADDRESS(0x0C)
+#define	SUE_SPI_REG_SER		SUE_SPI_BASEADDRESS(0x10)
+#define	SUE_SPI_REG_BAUDR	SUE_SPI_BASEADDRESS(0x14)
+#define	SUE_SPI_REG_TXFTLR	SUE_SPI_BASEADDRESS(0x18)
+#define	SUE_SPI_REG_RXFTLR	SUE_SPI_BASEADDRESS(0x1C)
+#define	SUE_SPI_REG_TXFLR	SUE_SPI_BASEADDRESS(0x20)
+#define	SUE_SPI_REG_RXFLR	SUE_SPI_BASEADDRESS(0x24)
+#define	SUE_SPI_REG_SR		SUE_SPI_BASEADDRESS(0x28)
+#define	SUE_SPI_REG_IMR		SUE_SPI_BASEADDRESS(0x2C)
+#define	SUE_SPI_REG_ISR		SUE_SPI_BASEADDRESS(0x30)
+#define	SUE_SPI_REG_RISR	SUE_SPI_BASEADDRESS(0x34)
+#define	SUE_SPI_REG_TXOICR	SUE_SPI_BASEADDRESS(0x38)
+#define	SUE_SPI_REG_RXOICR	SUE_SPI_BASEADDRESS(0x3C)
+#define	SUE_SPI_REG_RXUICR	SUE_SPI_BASEADDRESS(0x40)
+#define	SUE_SPI_REG_ICR		SUE_SPI_BASEADDRESS(0x48)
+#define	SUE_SPI_REG_DMACR	SUE_SPI_BASEADDRESS(0x4C)
+#define	SUE_SPI_REG_DMATDLR	SUE_SPI_BASEADDRESS(0x50)
+#define	SUE_SPI_REG_DMARDLR	SUE_SPI_BASEADDRESS(0x54)
+#define	SUE_SPI_REG_DR		SUE_SPI_BASEADDRESS(0x60)
+#define	SUE_SPI_REG_SPICTRLR0	SUE_SPI_BASEADDRESS(0xF4)
+
+extern const struct spi_ops spi_ops;
```

Comment:
I think this is declared in a platform header - check ssp

Reply:
@lgirdwood 
the "dai_ops" is declared in the ssp.h file.